tools/tsp-client/src/git.ts

```diff
@@ -0,0 +1,82 @@
+import { execSync, spawn } from "child_process";
+
+export function getRepoRoot(): string {
+    return execSync('git rev-parse --show-toplevel').toString().trim();
+}
+  
+export async function cloneRepo(rootUrl: string, cloneDir: string, repo: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const git = spawn("git", ["clone", "--no-checkout", "--filter=tree:0", repo, cloneDir], {
```

Comment:
seems like we could make a spawn helper since there's a lot of repeated code in this file

Also keep in mind any strings passed to spawn allow for shell injection attacks

Reply:
Tracking issue for future PR: https://github.com/Azure/azure-sdk-tools/issues/7046